staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -181,6 +196,20 @@ func (f *ConfigFlags) ToRawKubeConfigLoader() clientcmd.ClientConfig {
 	return clientConfig
 }
 
+// toRawKubePersistentConfigLoader binds config flag values to config overrides
+// Returns a persistent clientConfig for propagation.
+func (f *ConfigFlags) toRawKubePersistentConfigLoader() clientcmd.ClientConfig {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	if f.clientConfig == nil {
+		clientConfig := f.toRawKubeConfigLoader()
```

Comment:
No need for these two steps, assign right away to `f.clientConfig`.

Reply:
@soltysh We should make it thread safe. Please refer to the [comments](https://github.com/kubernetes/kubernetes/pull/71117#discussion_r234261548) by @liggitt @smarterclayton .

Reply:
There's no other place writing to `f.clientConfig` than this and you are already setting a lock to ensure you're the only one writing. That's sufficient. What Clayton and Jordan previously alluded to is that in your original version you didn't have any such mechanism in place.

Reply:
> assign right away to f.clientConfig

> There's no other place writing to `f.clientConfig` than this and you are already setting a lock to ensure you're the only one writing.

@soltysh We also want to load this config only once. Directly assigning to `clientConfig` means the config file will be parsed and reloaed everytime when invoked.

Reply:
> Directly assigning to clientConfig means the config file will be parsed and reloaed everytime when invoked.

How come? What I'm saying here is that the intermediate step of assigning to a local variable `clientConfig` is not necessary. You can just do `f.clientConfig = f.toRawKubeConfigLoader()` since this is thread safe (you're setting a lock at the beginning of this method) and you are assigning to `ConfigFlags` struct (no reloading happens).

Reply:
@soltysh Sorry, I misunderstood. Updated. PTAL.